compiler/executable/pipeline.rs

```diff
@@ -145,12 +147,14 @@ pub fn compile_stages_and_fetch(
     (HashMap<Variable, VariablePosition>, Vec<ExecutableStage>, Option<Arc<ExecutableFetch>>),
     ExecutableCompilationError,
 > {
+    let selected_variables = variable_registry.variable_names().keys().copied().collect_vec();
```

Comment:
We used to use all the variables down the pipeline. However, this is incorrect because we also pass anonymous variables, and later, we fail to find positions for them.
Here we introduce selected variables, that are expected to be used later.